libs/common/src/types/tfm/mapped-deal.ts

```diff
@@ -0,0 +1,67 @@
+import { ObjectId } from 'mongodb';
+import { DEAL_TYPE } from '../../constants';
+import { AnyObject } from '../any-object';
+import { DealSubmissionType, DealType, TfmActivity, TfmDealCancellationWithStatus } from '..';
+import { TfmDealStage } from './deal-stage';
+
+/**
+ * In TFM we map deals from the database to take on a different structure which is then passed around TFM-UI and TFM-API.
+ * these types are incomplete and should be added to as and when new properties are discovered
+ */
+
+type MappedBaseDealSnapshot = AnyObject & {
+  _id: ObjectId;
+  submissionType: DealSubmissionType;
+  isFinanceIncreasing: boolean;
+  details: {
+    ukefDealId: string;
+  };
+  facilities: AnyObject;
+  totals: AnyObject;
+  submissionDetails: AnyObject;
+  eligibility: AnyObject;
+};
+
+export interface MappedBssEwcsDealSnapshot extends MappedBaseDealSnapshot {
+  dealType: typeof DEAL_TYPE.BSS_EWCS;
+}
+
+export interface MappedGefDealSnapshot extends MappedBaseDealSnapshot {
+  dealType: typeof DEAL_TYPE.GEF;
+  status: string;
+  updatedAt: number;
+  maker: AnyObject;
+  bank: AnyObject;
+  exporter: {
+    companyName: string;
+  };
+  bankInternalRefName: string;
+  additionalRefName: string;
+  supportingInformation: {
+    status: string;
+  };
+}
+
+export type MappedDealSnapshot = MappedBssEwcsDealSnapshot | MappedGefDealSnapshot;
+
+export type MappedDealTfm = {
+  supplyContractValueInGBP: string | null;
+  dateReceived: string;
+  dateReceivedTimestamp: number;
+  product: DealType;
+  stage: TfmDealStage;
+  exporterCreditRating: string;
+  lastUpdated: number;
+  lossGivenDefault: number;
+  probabilityOfDefault: number;
+  activities: TfmActivity[];
+  parties: AnyObject;
+  cancellation?: TfmDealCancellationWithStatus;
+  tasks?: AnyObject[];
+};
+
+export type MappedDeal = {
```

Comment:
I've made these new types to reflect the mapped types in e.g. `trade-finance-manager-api/src/v1/rest-mappings/deal.js` and `mapDealSnapshot.js` - this will make it easier to make and use TS files in TFM-UI and TFM-API